core/rfb.js

```diff
@@ -1543,6 +1872,90 @@
             sock.flush();
         },
 
+        atenKeyEvent: function (sock, keysym, down) {
+            var ks = XK2HID[keysym];
```

Comment:
This looks like it assumes a US keyboard layout. You should look at the Qemu code on how to do this properly.